spec/requests/sessions_spec.rb

```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+
+describe 'Sessions', type: :request do
+  let(:recaptcha_id) { 'recaptcha_form' }
+
+  describe '#new' do
+    after { cookies[:login_recaptcha] = nil }
+
+    context 'when recaptcha is not configured' do
+      before do
+        allow(RecaptchaService).to receive(:recaptcha_configured?).and_return(false)
+      end
+
+      it 'does not display recaptcha' do
+        cookies[:login_recaptcha] = true
+        get new_user_session_path
+
+        expect(response.body).to_not include recaptcha_id
+      end
+    end 
+
+    context 'when recaptcha is configured' do
+      before do
+        allow(RecaptchaService).to receive(:recaptcha_configured?).and_return(true)
+      end
+
+      context 'when user has recaptcha cookie' do
+        it 'displays recaptcha' do
+          cookies[:login_recaptcha] = true
+          get new_user_session_path
+
+          expect(response.body).to include recaptcha_id
+        end
+      end
+
+      context 'when recaptcha is specified in query string' do
+        before do
+          get new_user_session_path, params: { recaptcha: true }
+        end
+
+        it 'displays recaptcha' do
+          expect(response.body).to include recaptcha_id
+        end
+
+        it 'sets recaptcha cookie' do
+          expect(cookies[:login_recaptcha]).to eq 'true'
```

Comment:
Should this be boolean instead of a string?

Reply:
That's what I thought, as well, but I was running into issues with it coming back as a string. I think cookies can only be stored as strings, which was why this was necessary, but I could be wrong.